tests-e2e/membership_transitions_test.go

```diff
@@ -659,6 +660,47 @@ func TestHeroesOnMembershipChanges(t *testing.T) {
 			t.Errorf("expected userID %q, got %q", gotUserID, bob.UserID)
 		}
 	})
+
+	t.Run("can set heroes=true in lists", func(t *testing.T) {
+		listRoomID := alice.CreateRoom(t, map[string]interface{}{"preset": "public_chat"})
+		bob.JoinRoom(t, listRoomID, []string{})
+
+		res := alice.SlidingSyncUntil(t, "", sync3.Request{
+			Lists: map[string]sync3.RequestList{
+				"all_rooms": {
+					Ranges: sync3.SliceRanges{{0, 20}},
+					RoomSubscription: sync3.RoomSubscription{
+						Heroes:        &boolTrue,
+						TimelineLimit: 10,
+					},
+				},
+			},
+		}, func(response *sync3.Response) error {
+			r, ok := response.Rooms[listRoomID]
+			if !ok {
+				return fmt.Errorf("room %q not in response", listRoomID)
+			}
+			// wait for bob to be joined
```

Comment:
Can't you just use `res := alice.SlidingSyncUntilMembership(t, "", listRoomID, bob, "join")`?

Reply:
For the previous test yes, but this test should use a list instead of a room subscription (used in `SlidingSyncUntilMembership`)